app/controllers/memberships_controller.rb

```diff
@@ -27,6 +27,13 @@ def approval
     redirect_to community_memberships_path(@membership.community.slug)
   end
 
+  def set_moderator
+    authorize Membership
+    @membership = Membership.find_by!(id: params[:membership_id])
+    @membership.set_moderator!
+    redirect_to community_memberships_path(@membership.community.slug)
+  end
```

Comment:
Not blocking - Instead of a one-off action, I would encourage a route like `/communities/zinc/moderators/:user_id` that when put, makes the provided  user a moderator, and when deleted removes the person's  moderator status.